src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -41,58 +58,189 @@ public EmailFormatter(String emailMgs, AttachmentHandler attachmentHandler) thro
         ObjectMapper mapper = new ObjectMapper();
         this.emailMessage = emailMgs;
         this.attachmentHandler = attachmentHandler;
-        this.messageMap = mapper.readValue(this.emailMessage, new TypeReference<HashMap<String, Object>>(){});
+        this.messageMap = mapper.readValue(this.emailMessage, new TypeReference<HashMap<String, Object>>() {
+        });
+    }
+
+    public List<Object> rfc822Format() {
+        List<Object> emailMsgPlusStatus = rfcFormatWithoutAttachmentLimitCheck();
+        String rfcFormattedText = (String) emailMsgPlusStatus.get(0);
+        JSONObject status = (JSONObject) emailMsgPlusStatus.get(1);
+        status.put(Utils.JSON_ATTR_ITEM_ID, getItemId());
+
+        // Something bad happened in formatting the email
+        if (rfcFormattedText == null) {
+            log.error("Email formatting is not successful");
+
+            status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_ERROR);
+            status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+            emailMsgPlusStatus.set(1, status);
+            return emailMsgPlusStatus;
+        }
+        // Large email size check, if grater than proposed limit attachment will be dropped
+        if (checkMsgSizeMoreThanExpectedLimit(rfcFormattedText)) {
+            log.warn("Attachments are dropped for the message");
+
+            rfcFormattedText = removeAttachments(rfcFormattedText);
+            emailMsgPlusStatus.set(0, rfcFormattedText);
+            status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_PARTIAL);
+            status.put(Utils.JSON_ATTR_MESSAGE, "email Message size exceeds the expected limit, attachments dropped");
```

Comment:
I spoke to the BSA, they agreed to drop all the attachment not go into path if 2 attachment > 16mb vs 3 attachment > 16mb route when entire message including attachment found > 16mb